src/Logic/DirectionalLight.java

```diff
@@ -0,0 +1,38 @@
+package Logic;
+
+public class DirectionalLight extends Light
+{
+	public DirectionalLight()
+	{
+		super();
+	}
+	
+	public DirectionalLight(float directionX, float directionY, float directionZ,  float colorR, float colorG, float colorB)
+	{
+		super(colorR, colorG, colorB);
+		this.directionX = directionX;
+		this.directionY = directionY;
+		this.directionZ = directionZ;
+	}
+	
+	public float directionX = 0;
```

Comment:
[Fix] ordinea in java este: campuri, constructori, getteri + setteri

la fel si pentru restul claselor tale